WordPress/gutenberg
# `RadioGroup`: migrate from `reakit` to `ariakit`

## What?

- Migrate `RadioGroup` from `reakit` to `ariakit`
- Migrate to TypeScript

## Why?

Part of #53278

As stated in the linked issue [Reakit](https://reakit.io/) does not explicitly support React 18 (causing peer dependency warnings) and Reakit has already been succeeded by [Ariakit](https://ariakit.org/), so it will not receive any more updates.

## How?

We switch out the Reakit dependency for Ariakit.

## Testing Instructions

- There shouldn't be any type errors
- Verify the component still looks and works as expected (verify via Storybook)

Please note, this component is deprecated and the sole purpose for this PR is migrating from `reakit` to `Ariakit`. Besides that we maintain the existing API to prevent breaking changes.
